contracts/KYCRecordRegistry2.sol

```diff
@@ -0,0 +1,163 @@
+pragma solidity ^0.8.7;
+pragma abicoder v2;
+
+// OpenZeppelin v4
+import { Initializable } from "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+
+import { SNARK_SCALAR_FIELD } from "./helpers/Globals.sol";
+
+import { PoseidonT3 } from "./helpers/Poseidon.sol";
+
+import { KYCCenterRegistry } from "./KYCCenterRegistry.sol";
+
+/**
+ * @title KYCRecordRegistry
+ * @author Galactica dev team
+ * @notice Batch Incremental Merkle Tree for zkKYC records
+ * Relevant external contract calls should be in those functions, not here
+ */
+contract KYCRecordRegistry2 is Initializable {
+  // NOTE: The order of instantiation MUST stay the same across upgrades
+  // add new variables to the bottom of the list and decrement the __gap
+  // variable at the end of this file
+  // See https://docs.openzeppelin.com/learn/upgrading-smart-contracts#upgrading
+
+  // The tree depth and size
+  uint256 internal constant TREE_DEPTH = 32;
+  uint256 internal constant TREE_SIZE = 2 ** 32;
+
+  // Tree zero value
+  bytes32 public constant ZERO_VALUE = bytes32(uint256(keccak256("Galactica")) % SNARK_SCALAR_FIELD);
+
+  // Next leaf index (number of inserted leaves in the current tree)
+  uint256 public nextLeafIndex;
+
+  // The Merkle root
+  bytes32 public merkleRoot;
+
+  // The Merkle path to the leftmost leaf upon initialization. It *should
+  // not* be modified after it has been set by the initialize function.
+  // Caching these values is essential to efficient appends.
+  bytes32[TREE_DEPTH] public zeros;
+
+
+  KYCCenterRegistry public _KYCCenterRegistry;
+  event zkKYCRecordAddition(bytes32 indexed zkKYCRecordLeafHash, address indexed KYCCenter);
+  event zkKYCRecordRevocation(bytes32 indexed zkKYCRecordLeafHash, address indexedKYCCenter);
+
+  /**
+   * @notice Calculates initial values for Merkle Tree
+   * @dev OpenZeppelin initializer ensures this can only be called once
+   */
+  function initializeKYCRecordRegistry(address KYCCenterRegistry_) internal onlyInitializing {
+    /*
+    To initialize the Merkle tree, we need to calculate the Merkle root
+    assuming that each leaf is the zero value.
+    H(H(a,b), H(c,d))
+      /          \
+    H(a,b)     H(c,d)
+    /   \       /  \
+    a    b     c    d
+    `zeros` and `filledSubTrees` will come in handy later when we do
+    inserts or updates. e.g when we insert a value in index 1, we will
+    need to look up values from those arrays to recalculate the Merkle
+    root.
+    */
+
+    // Calculate zero values
+    zeros[0] = ZERO_VALUE;
+
+    // Store the current zero value for the level we just calculated it for
+    bytes32 currentZero = ZERO_VALUE;
+
+    // Loop through each level
+    for (uint256 i = 0; i < TREE_DEPTH; i += 1) {
+      // Push it to zeros array
+      zeros[i] = currentZero;
+
+      // Set filled subtrees to a value so users don't pay storage allocation costs
+      filledSubTrees[i] = currentZero;
```

Comment:
This variable is not declared.

Reply:
I removed it, because we don't need it in the new version